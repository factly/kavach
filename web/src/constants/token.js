//Actions
export const ADD_TOKEN = 'ADD_TOKEN';
export const ADD_TOKENS = 'ADD_TOKENS';
export const ADD_TOKENS_REQUEST = 'ADD_TOKENS_REQUEST';
export const SET_TOKENS_LOADING = 'SET_TOKENS_LOADING';
export const RESET_TOKENS = 'RESET_TOKENS';

//API
export const TOKENS_API = '/tokens';
