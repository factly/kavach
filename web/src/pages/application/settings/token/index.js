import React from 'react';
import { Skeleton, Space, Button } from 'antd';
import TokenList from './components/TokenList';
import { useSelector } from 'react-redux';
import { Link, useParams } from 'react-router-dom';

export default function ApplicationTokens() {
  const { id } = useParams();
  const { application, loadingApp, role, loading } = useSelector((state) => {
    return {
      application: state.applications.details[id],
      loadingApp: state.applications.loading,
      role: state.profile.roles[state.organisations.selected],
      loading: state.profile.loading,
    };
  });

  return (
    <Space direction="vertical">
      <Link key="1" to={`/applications/${id}/settings`}>
        <Button type="primary"> Back to Settings </Button>
      </Link>
      <h2>Tokens in {application.name}</h2>
      {loading || loadingApp ? (
        <Skeleton />
      ) : (
        <Space direction="vertical">
          {role === 'owner' ? (
            <div style={{ display: 'flex', justifyContent: 'end' }}>
              <Link
                key="1"
                to={{
                  pathname: `/applications/${id}/settings/tokens/create`,
                }}
              >
                <Button type="primary"> Generate new tokens </Button>
              </Link>
            </div>
          ) : null}
          <TokenList appID={id} application={application} role={role} />
        </Space>
      )}
    </Space>
  );
}
