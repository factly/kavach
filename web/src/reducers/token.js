import {
  ADD_APPLICATION_TOKENS,
  DELETE_APPLICATION_TOKEN,
  ADD_ORGANISATION_TOKENS,
  DELETE_ORGANISATION_TOKEN,
  ADD_SPACE_TOKENS,
  DELETE_SPACE_TOKEN,
  SET_TOKENS_LOADING,
} from '../constants/token';

const initialState = {
  organisations: [],
  applications: [],
  spaces: [],
  loading: true,
};

export default function tokensReducer(state = initialState, action = {}) {
  switch (action.type) {
    case ADD_APPLICATION_TOKENS:
      return {
        ...state,
        applications: action.payload,
      };
    case DELETE_APPLICATION_TOKEN:
      return {
        ...state,
        applications: state.applications.filter((token) => token.id !== action.payload),
      };
    case ADD_ORGANISATION_TOKENS:
      return {
        ...state,
        organisations: action.payload,
      };
    case DELETE_ORGANISATION_TOKEN:
      return {
        ...state,
        organisations: state.organisations.filter((token) => token.id !== action.payload),
      };
    case ADD_SPACE_TOKENS:
      return {
        ...state,
        spaces: action.payload,
      };
    case DELETE_SPACE_TOKEN:
      return {
        ...state,
        spaces: state.spaces.filter((token) => token.id !== action.payload),
      };
    case SET_TOKENS_LOADING:
      return {
        ...state,
        loading: action.payload,
      };
    default:
      return state;
  }
}
