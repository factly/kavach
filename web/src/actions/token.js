import axios from 'axios';
import { ORGANISATIONS_API } from '../constants/organisations';
import {
  ADD_TOKEN,
  ADD_TOKENS,
  ADD_TOKENS_REQUEST,
  SET_TOKENS_LOADING,
  RESET_TOKENS,
} from '../constants/token';
import { addErrorNotification, addSuccessNotification } from './notifications';

export const addToken = (data, appID) => {
  return (dispatch, getState) => {
    dispatch(loadingTokens());
    return axios
      .post(ORGANISATIONS_API + '/' + getState().organisations.selected + '/applications/' + appID + '/tokens', data)
      .then((res) => {
        dispatch(resetTokens());
        dispatch(addSuccessNotification('Token Added')) 
        return res.data
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};

export const deleteToken = (id, appID) => {
  return (dispatch, getState) => {
    dispatch(loadingTokens());
    return axios
      .delete(ORGANISATIONS_API + '/' + getState().organisations.selected + '/applications/' + appID + '/tokens/' + id)
      .then(() => {
        dispatch(resetTokens());
        dispatch(addSuccessNotification('Token Deleted')) 
      })
      .catch((error) => {
        dispatch(addErrorNotification(error.message));
      });
  };
};


export const loadingTokens = () => ({
  type: SET_TOKENS_LOADING,
  payload: true,
});

export const stopTokenLoading = () => ({
  type: SET_TOKENS_LOADING,
  payload: false,
});

export const getTokenByID = (data) => ({
  type: ADD_TOKEN,
  payload: data,
});

export const addTokensList = (data) => ({
  type: ADD_TOKENS,
  payload: data,
});

export const addTokensRequest = (data) => ({
  type: ADD_TOKENS_REQUEST,
  payload: data,
});

export const resetTokens = () => ({
  type: RESET_TOKENS,
});