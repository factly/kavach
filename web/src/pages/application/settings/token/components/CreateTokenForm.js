import React from 'react';
import { Form, Input, Button, Card, Skeleton, Modal } from 'antd';
import { addApplicationToken } from '../../../../../actions/token';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useHistory, useParams } from 'react-router-dom';
import ErrorComponent from '../../../../../components/ErrorsAndImage/ErrorComponent';
import { getApplication } from '../../../../../actions/application';
import ClipBoardCopy from '../../../../../utils/clipboardClick';

const tailLayout = {
  wrapperCol: {
    offset: 8,
  },
};

const CreateApplicationTokenForm = () => {
  const dispatch = useDispatch();
  const { id } = useParams();
  const [form] = Form.useForm();
  const history = useHistory();
  const { TextArea } = Input;
  const [token, setToken] = React.useState(null);
  const [showModal, setShowModal] = React.useState(false);
  const onReset = () => {
    form.resetFields();
  };

  const onCreate = (values) => {
    dispatch(addApplicationToken(id, values, setToken, setShowModal));
  };

  const handleOk = () => {
    history.push(`/applications/${id}/settings/tokens`);
  };

  const { application, loadingApp, role, loadingRole } = useSelector((state) => {
    return {
      application: state.applications.details[id] ? state.applications.details[id] : null,
      loadingApps: state.applications.loading,
      role: state.profile.roles[state.organisations.selected],
      loadingRole: state.profile.loading,
    };
  });

  React.useEffect(() => {
    dispatch(getApplication(id));
  }, [dispatch, id]);

  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        gap: '20px',
      }}
    >
      <Link key="1" to={`/applications/${id}/settings/tokens`}>
        <Button type="primary"> Back to Tokens</Button>
      </Link>
      {loadingApp || loadingRole ? (
        <Skeleton />
      ) : role !== 'owner' ? (
        <ErrorComponent
          status="403"
          title="Sorry you are not authorised to access this page"
          link="/organisation"
          message="Back Home"
        />
      ) : (
        <Card
          title={`Create Application Token - ${application?.name}`}
          style={{
            width: '50%',
            alignSelf: 'center',
          }}
        >
          <Form
            form={form}
            layout="vertical"
            name="create-application-token"
            onFinish={(values) => {
              onCreate(values);
              onReset();
            }}
          >
            <Form.Item
              name="application_name"
              label="Application Name"
              initialValue={application.name}
            >
              <Input disabled={true} />
            </Form.Item>
            <Form.Item
              name="name"
              label="Name"
              rules={[
                {
                  required: true,
                  message: 'Please enter the name!',
                },
                { min: 3, message: 'Name must be minimum 3 characters.' },
                { max: 50, message: 'Name must be maximum 50 characters.' },
              ]}
            >
              <Input />
            </Form.Item>
            <Form.Item
              name="description"
              label="Description"
              rules={[
                {
                  required: true,
                  message: 'Please input token description!',
                },
              ]}
            >
              <TextArea rows={4} />
            </Form.Item>
            <Form.Item {...tailLayout}>
              <Button type="primary" htmlType="submit">
                Generate Token
              </Button>
            </Form.Item>
          </Form>
          <Modal
            title="Generated Token"
            visible={showModal}
            width={650}
            closable={false}
            okText="Goto Tokens"
            onOk={handleOk}
            cancelButtonProps={{
              style: {
                display: 'none',
              },
            }}
          >
            <ClipBoardCopy text={showModal === true ? token : ''} />
          </Modal>
        </Card>
      )}
    </div>
  );
};

export default CreateApplicationTokenForm;
