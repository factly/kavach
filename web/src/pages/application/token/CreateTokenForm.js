import React, { useState } from 'react';
import { Form, Input, Button } from 'antd';
import { addToken } from '../../../actions/token';
import { useDispatch } from 'react-redux';

const layout = {
  labelCol: {
    span: 7,
  },
  wrapperCol: {
    span: 12,
  },
};
const tailLayout = {
  wrapperCol: {
    offset: 8,
  },
};

const CreateTokenForm = ({ appID, setVisible, setTokenFlag }) => {
  const dispatch = useDispatch();
  const [form] = Form.useForm();
  const { TextArea } = Input;

  const onReset = () => {
    form.resetFields();
  };

  const onCreate = (values) => {
    dispatch(addToken(values, appID)).then(() => {
      setVisible(false);
      setTokenFlag(true);
    });
  };

  return (
    <Form
      form={form}
      layout="vertical"
      name="create-token"
      onFinish={(values) => {
        onCreate(values);
        onReset();
      }}
    >
      <Form.Item
        name="name"
        label="Name"
        rules={[
          {
            required: true,
            message: 'Please enter the name!',
          },
          { min: 3, message: 'Name must be minimum 3 characters.' },
          { max: 50, message: 'Name must be maximum 50 characters.' },
        ]}
      >
        <Input />
      </Form.Item>

      <Form.Item name="description" label="Description">
        <TextArea rows={4} />
      </Form.Item>
      <Form.Item {...tailLayout}>
        <Button type="primary" htmlType="submit">
          Create API Token
        </Button>
      </Form.Item>
    </Form>
  );
};

export default CreateTokenForm;
