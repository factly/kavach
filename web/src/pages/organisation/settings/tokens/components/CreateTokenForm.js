import React from 'react';
import { Form, Input, Button, Card, Skeleton, Modal } from 'antd';
import { addOrganisationToken } from '../../../../../actions/token';
import { useDispatch, useSelector } from 'react-redux';
import { Link, useHistory, useParams } from 'react-router-dom';
import ErrorComponent from '../../../../../components/ErrorsAndImage/ErrorComponent';
import { getOrganisation } from '../../../../../actions/organisations';
import ClipBoardCopy from '../../../../../utils/clipboardClick';

const tailLayout = {
  wrapperCol: {
    offset: 0,
    span: 5,
  },
};

const CreateOrganisationToken = () => {
  const dispatch = useDispatch();
  const { orgID } = useParams();
  const [form] = Form.useForm();
  const history = useHistory();
  const [token, setToken] = React.useState(null);
  const [showModal, setShowModal] = React.useState(false);
  const { TextArea } = Input;
  const onReset = () => {
    form.resetFields();
  };

  const onCreate = (values) => {
    dispatch(addOrganisationToken(values, setToken, setShowModal));
  };

  const { organisation, loadingOrg, role, loadingRole } = useSelector((state) => {
    return {
      organisation: state.organisations.details[orgID] ? state.organisations.details[orgID] : null,
      loadingApps: state.organisations.loading,
      role: state.profile.roles[state.organisations.selected],
      loadingRole: state.profile.loading,
    };
  });

  const handleOk = () => {
    history.push(`/organisation/${orgID}/settings/tokens`);
  };

  React.useEffect(() => {
    dispatch(getOrganisation(orgID));
  }, [dispatch, orgID]);

  return (
    <>
      {loadingOrg || loadingRole ? (
        <Skeleton />
      ) : role === 'owner' ? (
        <>
          <div className="organisation-descriptions-header">
            <div className="organisation-descriptions-title">
              <h2 className="organisation-title-main">
                Create Organisation Token - {organisation?.title}
              </h2>
            </div>
          </div>
          <Form
            form={form}
            layout="vertical"
            name="create-organisation-token"
            onFinish={(values) => {
              onCreate(values);
              onReset();
            }}
            style={{
              maxWidth: '600px',
            }}
          >
            <Form.Item
              name="name"
              label="Name"
              rules={[
                {
                  required: true,
                  message: 'Please enter the name!',
                },
                { min: 3, message: 'Name must be minimum 3 characters.' },
                { max: 50, message: 'Name must be maximum 50 characters.' },
              ]}
            >
              <Input />
            </Form.Item>
            <Form.Item
              name="description"
              label="Description"
              rules={[
                {
                  required: true,
                  message: 'Please input token description!',
                },
              ]}
            >
              <TextArea rows={4} />
            </Form.Item>
            <Form.Item {...tailLayout}>
              <Button type="primary" htmlType="submit" form="create-organisation-token">
                Generate Token
              </Button>
            </Form.Item>
          </Form>
          <Modal
            title="Generated Token"
            open={showModal}
            width={650}
            closable={false}
            okText="Goto Tokens"
            onOk={handleOk}
            cancelButtonProps={{
              style: {
                display: 'none',
              },
            }}
          >
            <ClipBoardCopy text={showModal === true ? token : ''} />
          </Modal>
        </>
      ) : (
        <ErrorComponent
          status="403"
          title="Sorry you are not authorised to access this page"
          link="/organisation"
          message="Back Home"
        />
      )}
    </>
  );
};

export default CreateOrganisationToken;
