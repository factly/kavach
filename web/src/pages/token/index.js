import React from 'react';
import { Tabs, Form } from 'antd';
import FormComponent from './components/TokenForm';
import TokenList from './components/TokenList';
import { addOrganisationToken, getOrganisationTokens } from '../../actions/token';
import { useDispatch } from 'react-redux';

function Token() {
  const dispatch = useDispatch();
  const [form] = Form.useForm();
  const onSubmit = (values, type) => {
    switch (type) {
      case 'organisation':
        dispatch(addOrganisationToken(values)).then(() => getOrganisationTokens());
        break;
      default:
        return;
    }
    form.resetFields();
  };

  return (
    <Tabs centered={true}>
      <Tabs.TabPane tab="Organisation Token" key="1">
        <FormComponent type="organisation" onSubmit={onSubmit} form={form} />
        <TokenList type="organisation" />
      </Tabs.TabPane>
      <Tabs.TabPane tab="Application Token" key="2">
        <FormComponent type="application" onSubmit={onSubmit} form={form} />
        <TokenList type="application" />
      </Tabs.TabPane>
      <Tabs.TabPane tab="Space Token" key="3">
        <FormComponent type="space" onSubmit={onSubmit} form={form} />
        <TokenList type="space" />
      </Tabs.TabPane>
    </Tabs>
  );
}

export default Token;
