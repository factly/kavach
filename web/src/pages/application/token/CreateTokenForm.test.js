import React from 'react';
import { BrowserRouter as Router, useHistory } from 'react-router-dom';
import { useDispatch, Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { act } from '@testing-library/react';
import { Button, Popconfirm, Table } from 'antd';
import { CloseOutlined } from '@ant-design/icons';

import '../../../matchMedia.mock';
import CreateTokenForm from './CreateTokenForm';
import { addToken } from '../../../actions/token';
const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn(),
}));
jest.mock('../../../actions/token', () => ({
  addToken: jest.fn(),
}));
let store;
let mockedDispatch;

describe('Create Token form component', () => {
  store = mockStore({
    applications: {
      req: [],
      details: {},
      loading: true,
    },
  });
  store.dispatch = jest.fn(() => ({}));
  mockedDispatch = jest.fn(() => Promise.resolve({}));
  useDispatch.mockReturnValue(mockedDispatch);
  describe('snapshot testing', () => {
    it('should render the component', () => {
      const tree = mount(
        <Provider store={store}>
          <CreateTokenForm />
        </Provider>,
      );
      expect(tree).toMatchSnapshot();
    });
  });
  describe('component testing', () => {
    let wrapper, props;
    beforeEach(() => {
      props = {
        appID: 1,
        setVisible: jest.fn(),
        setTokenFlag: jest.fn(),
      };
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CreateTokenForm {...props} />
          </Provider>,
        );
      });
    });
    afterEach(() => {
      wrapper.unmount();
    });
    it('should not submit form with empty data', (done) => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CreateTokenForm appID={1} setVisible={jest.fn()} setTokenFlag={jest.fn()} />
          </Provider>,
        );
      });

      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Create API Token');
        submitButtom.simulate('submit');
        wrapper.update();
      });

      setTimeout(() => {
        expect(addToken).not.toHaveBeenCalled();
        done();
      }, 0);
    });
    it('should not submit form with data', (done) => {
      act(() => {
        wrapper = mount(
          <Provider store={store}>
            <CreateTokenForm appID={1} setVisible={jest.fn()} setTokenFlag={jest.fn()} />
          </Provider>,
        );
      });
      act(() => {
        wrapper
          .find('FormItem')
          .at(0)
          .find('Input')
          .simulate('change', { target: { value: 'Token Name' } });
        wrapper
          .find('FormItem')
          .at(1)
          .find('TextArea')
          .simulate('change', { target: { value: 'New Description' } });
      });
      act(() => {
        const submitButtom = wrapper.find('Button').at(0);
        expect(submitButtom.text()).toBe('Create API Token');
        submitButtom.simulate('submit');
        wrapper.update();
      });

      setTimeout(() => {
        expect(addToken).toHaveBeenCalledWith(
          {
            name: 'Token Name',
            description: 'New Description',
          },
          1,
        );
        done();
      }, 0);
    });
  });
});
