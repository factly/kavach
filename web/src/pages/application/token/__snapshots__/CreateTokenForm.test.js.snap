// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Token form component snapshot testing should render the component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [MockFunction],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <CreateTokenForm>
    <ForwardRef(InternalForm)
      form={
        Object {
          "__INTERNAL__": Object {
            "itemRef": [Function],
            "name": "create-token",
          },
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "getInternalHooks": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "resetFields": [Function],
          "scrollToField": [Function],
          "setFields": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
        }
      }
      layout="vertical"
      name="create-token"
      onFinish={[Function]}
    >
      <SizeContextProvider>
        <ForwardRef(Form)
          className="ant-form ant-form-vertical"
          form={
            Object {
              "__INTERNAL__": Object {
                "itemRef": [Function],
                "name": "create-token",
              },
              "getFieldError": [Function],
              "getFieldInstance": [Function],
              "getFieldValue": [Function],
              "getFieldsError": [Function],
              "getFieldsValue": [Function],
              "getInternalHooks": [Function],
              "isFieldTouched": [Function],
              "isFieldValidating": [Function],
              "isFieldsTouched": [Function],
              "isFieldsValidating": [Function],
              "resetFields": [Function],
              "scrollToField": [Function],
              "setFields": [Function],
              "setFieldsValue": [Function],
              "submit": [Function],
              "validateFields": [Function],
            }
          }
          id="create-token"
          name="create-token"
          onFinish={[Function]}
          onFinishFailed={[Function]}
        >
          <form
            className="ant-form ant-form-vertical"
            id="create-token"
            onSubmit={[Function]}
          >
            <FormItem
              label="Name"
              name="name"
              rules={
                Array [
                  Object {
                    "message": "Please enter the name!",
                    "required": true,
                  },
                  Object {
                    "message": "Name must be minimum 3 characters.",
                    "min": 3,
                  },
                  Object {
                    "max": 50,
                    "message": "Name must be maximum 50 characters.",
                  },
                ]
              }
            >
              <WrapperField
                label="Name"
                messageVariables={
                  Object {
                    "label": "Name",
                  }
                }
                name="name"
                onReset={[Function]}
                rules={
                  Array [
                    Object {
                      "message": "Please enter the name!",
                      "required": true,
                    },
                    Object {
                      "message": "Name must be minimum 3 characters.",
                      "min": 3,
                    },
                    Object {
                      "max": 50,
                      "message": "Name must be maximum 50 characters.",
                    },
                  ]
                }
                trigger="onChange"
                validateTrigger="onChange"
              >
                <Field
                  fieldContext={
                    Object {
                      "__INTERNAL__": Object {
                        "itemRef": [Function],
                        "name": "create-token",
                      },
                      "getFieldError": [Function],
                      "getFieldInstance": [Function],
                      "getFieldValue": [Function],
                      "getFieldsError": [Function],
                      "getFieldsValue": [Function],
                      "getInternalHooks": [Function],
                      "isFieldTouched": [Function],
                      "isFieldValidating": [Function],
                      "isFieldsTouched": [Function],
                      "isFieldsValidating": [Function],
                      "resetFields": [Function],
                      "scrollToField": [Function],
                      "setFields": [Function],
                      "setFieldsValue": [Function],
                      "submit": [Function],
                      "validateFields": [Function],
                      "validateTrigger": "onChange",
                    }
                  }
                  key="_name"
                  label="Name"
                  messageVariables={
                    Object {
                      "label": "Name",
                    }
                  }
                  name={
                    Array [
                      "name",
                    ]
                  }
                  onReset={[Function]}
                  rules={
                    Array [
                      Object {
                        "message": "Please enter the name!",
                        "required": true,
                      },
                      Object {
                        "message": "Name must be minimum 3 characters.",
                        "min": 3,
                      },
                      Object {
                        "max": 50,
                        "message": "Name must be maximum 50 characters.",
                      },
                    ]
                  }
                  trigger="onChange"
                  validateTrigger="onChange"
                  valuePropName="value"
                >
                  <Row
                    className="ant-form-item"
                    key="row"
                  >
                    <div
                      className="ant-row ant-form-item"
                      style={Object {}}
                    >
                      <FormItemLabel
                        htmlFor="create-token_name"
                        label="Name"
                        name="name"
                        prefixCls="ant-form"
                        required={true}
                        requiredMark={true}
                        rules={
                          Array [
                            Object {
                              "message": "Please enter the name!",
                              "required": true,
                            },
                            Object {
                              "message": "Name must be minimum 3 characters.",
                              "min": 3,
                            },
                            Object {
                              "max": 50,
                              "message": "Name must be maximum 50 characters.",
                            },
                          ]
                        }
                      >
                        <Col
                          className="ant-form-item-label"
                        >
                          <div
                            className="ant-col ant-form-item-label"
                            style={Object {}}
                          >
                            <label
                              className="ant-form-item-required"
                              htmlFor="create-token_name"
                              title="Name"
                            >
                              Name
                            </label>
                          </div>
                        </Col>
                      </FormItemLabel>
                      <FormItemInput
                        errors={Array []}
                        label="Name"
                        name={
                          Array [
                            "name",
                          ]
                        }
                        onDomErrorVisibleChange={[Function]}
                        prefixCls="ant-form"
                        rules={
                          Array [
                            Object {
                              "message": "Please enter the name!",
                              "required": true,
                            },
                            Object {
                              "message": "Name must be minimum 3 characters.",
                              "min": 3,
                            },
                            Object {
                              "max": 50,
                              "message": "Name must be maximum 50 characters.",
                            },
                          ]
                        }
                        status=""
                        touched={false}
                        validateStatus=""
                        validating={false}
                      >
                        <Col
                          className="ant-form-item-control"
                        >
                          <div
                            className="ant-col ant-form-item-control"
                            style={Object {}}
                          >
                            <div
                              className="ant-form-item-control-input"
                            >
                              <div
                                className="ant-form-item-control-input-content"
                              >
                                <Component
                                  update={1}
                                >
                                  <Input
                                    id="create-token_name"
                                    onChange={[Function]}
                                    type="text"
                                  >
                                    <ClearableLabeledInput
                                      bordered={true}
                                      element={
                                        <input
                                          className="ant-input"
                                          id="create-token_name"
                                          onBlur={[Function]}
                                          onChange={[Function]}
                                          onFocus={[Function]}
                                          onKeyDown={[Function]}
                                          type="text"
                                        />
                                      }
                                      focused={false}
                                      handleReset={[Function]}
                                      id="create-token_name"
                                      inputType="input"
                                      onChange={[Function]}
                                      prefixCls="ant-input"
                                      triggerFocus={[Function]}
                                      type="text"
                                      value=""
                                    >
                                      <input
                                        className="ant-input"
                                        id="create-token_name"
                                        onBlur={[Function]}
                                        onChange={[Function]}
                                        onFocus={[Function]}
                                        onKeyDown={[Function]}
                                        type="text"
                                        value=""
                                      />
                                    </ClearableLabeledInput>
                                  </Input>
                                </Component>
                              </div>
                            </div>
                            <ErrorList
                              errors={Array []}
                              onDomErrorVisibleChange={[Function]}
                            >
                              <CSSMotion
                                motionAppear={true}
                                motionDeadline={500}
                                motionName="show-help"
                                onLeaveEnd={[Function]}
                                removeOnLeave={true}
                                visible={false}
                              >
                                <DomWrapper />
                              </CSSMotion>
                            </ErrorList>
                          </div>
                        </Col>
                      </FormItemInput>
                    </div>
                  </Row>
                </Field>
              </WrapperField>
            </FormItem>
            <FormItem
              label="Description"
              name="description"
            >
              <WrapperField
                label="Description"
                messageVariables={
                  Object {
                    "label": "Description",
                  }
                }
                name="description"
                onReset={[Function]}
                trigger="onChange"
                validateTrigger="onChange"
              >
                <Field
                  fieldContext={
                    Object {
                      "__INTERNAL__": Object {
                        "itemRef": [Function],
                        "name": "create-token",
                      },
                      "getFieldError": [Function],
                      "getFieldInstance": [Function],
                      "getFieldValue": [Function],
                      "getFieldsError": [Function],
                      "getFieldsValue": [Function],
                      "getInternalHooks": [Function],
                      "isFieldTouched": [Function],
                      "isFieldValidating": [Function],
                      "isFieldsTouched": [Function],
                      "isFieldsValidating": [Function],
                      "resetFields": [Function],
                      "scrollToField": [Function],
                      "setFields": [Function],
                      "setFieldsValue": [Function],
                      "submit": [Function],
                      "validateFields": [Function],
                      "validateTrigger": "onChange",
                    }
                  }
                  key="_description"
                  label="Description"
                  messageVariables={
                    Object {
                      "label": "Description",
                    }
                  }
                  name={
                    Array [
                      "description",
                    ]
                  }
                  onReset={[Function]}
                  trigger="onChange"
                  validateTrigger="onChange"
                  valuePropName="value"
                >
                  <Row
                    className="ant-form-item"
                    key="row"
                  >
                    <div
                      className="ant-row ant-form-item"
                      style={Object {}}
                    >
                      <FormItemLabel
                        htmlFor="create-token_description"
                        label="Description"
                        name="description"
                        prefixCls="ant-form"
                        required={false}
                        requiredMark={true}
                      >
                        <Col
                          className="ant-form-item-label"
                        >
                          <div
                            className="ant-col ant-form-item-label"
                            style={Object {}}
                          >
                            <label
                              className=""
                              htmlFor="create-token_description"
                              title="Description"
                            >
                              Description
                            </label>
                          </div>
                        </Col>
                      </FormItemLabel>
                      <FormItemInput
                        errors={Array []}
                        label="Description"
                        name={
                          Array [
                            "description",
                          ]
                        }
                        onDomErrorVisibleChange={[Function]}
                        prefixCls="ant-form"
                        status=""
                        touched={false}
                        validateStatus=""
                        validating={false}
                      >
                        <Col
                          className="ant-form-item-control"
                        >
                          <div
                            className="ant-col ant-form-item-control"
                            style={Object {}}
                          >
                            <div
                              className="ant-form-item-control-input"
                            >
                              <div
                                className="ant-form-item-control-input-content"
                              >
                                <Component
                                  update={1}
                                >
                                  <ForwardRef
                                    id="create-token_description"
                                    onChange={[Function]}
                                    rows={4}
                                  >
                                    <ClearableLabeledInput
                                      bordered={true}
                                      element={
                                        <TextArea
                                          className=""
                                          id="create-token_description"
                                          onChange={[Function]}
                                          prefixCls="ant-input"
                                          rows={4}
                                        />
                                      }
                                      handleReset={[Function]}
                                      id="create-token_description"
                                      inputType="text"
                                      onChange={[Function]}
                                      prefixCls="ant-input"
                                      rows={4}
                                      value=""
                                    >
                                      <TextArea
                                        className=""
                                        id="create-token_description"
                                        onChange={[Function]}
                                        prefixCls="ant-input"
                                        rows={4}
                                        value=""
                                      >
                                        <ResizableTextArea
                                          className=""
                                          id="create-token_description"
                                          onChange={[Function]}
                                          onKeyDown={[Function]}
                                          prefixCls="ant-input"
                                          rows={4}
                                          value=""
                                        >
                                          <ResizeObserver
                                            disabled={true}
                                            onResize={[Function]}
                                          >
                                            <textarea
                                              className="ant-input"
                                              id="create-token_description"
                                              onChange={[Function]}
                                              onKeyDown={[Function]}
                                              rows={4}
                                              style={Object {}}
                                              value=""
                                            />
                                          </ResizeObserver>
                                        </ResizableTextArea>
                                      </TextArea>
                                    </ClearableLabeledInput>
                                  </ForwardRef>
                                </Component>
                              </div>
                            </div>
                            <ErrorList
                              errors={Array []}
                              onDomErrorVisibleChange={[Function]}
                            >
                              <CSSMotion
                                motionAppear={true}
                                motionDeadline={500}
                                motionName="show-help"
                                onLeaveEnd={[Function]}
                                removeOnLeave={true}
                                visible={false}
                              >
                                <DomWrapper />
                              </CSSMotion>
                            </ErrorList>
                          </div>
                        </Col>
                      </FormItemInput>
                    </div>
                  </Row>
                </Field>
              </WrapperField>
            </FormItem>
            <FormItem
              wrapperCol={
                Object {
                  "offset": 8,
                }
              }
            >
              <Row
                className="ant-form-item"
                key="row"
              >
                <div
                  className="ant-row ant-form-item"
                  style={Object {}}
                >
                  <FormItemLabel
                    prefixCls="ant-form"
                    requiredMark={true}
                    wrapperCol={
                      Object {
                        "offset": 8,
                      }
                    }
                  />
                  <FormItemInput
                    errors={Array []}
                    onDomErrorVisibleChange={[Function]}
                    prefixCls="ant-form"
                    status=""
                    validateStatus=""
                    wrapperCol={
                      Object {
                        "offset": 8,
                      }
                    }
                  >
                    <Col
                      className="ant-form-item-control"
                      offset={8}
                    >
                      <div
                        className="ant-col ant-col-offset-8 ant-form-item-control"
                        style={Object {}}
                      >
                        <div
                          className="ant-form-item-control-input"
                        >
                          <div
                            className="ant-form-item-control-input-content"
                          >
                            <Button
                              htmlType="submit"
                              type="primary"
                            >
                              <Wave>
                                <button
                                  className="ant-btn ant-btn-primary"
                                  onClick={[Function]}
                                  type="submit"
                                >
                                  <LoadingIcon
                                    existIcon={false}
                                    loading={false}
                                    prefixCls="ant-btn"
                                  >
                                    <CSSMotion
                                      motionName="ant-btn-loading-icon-motion"
                                      onAppearActive={[Function]}
                                      onAppearStart={[Function]}
                                      onEnterActive={[Function]}
                                      onEnterStart={[Function]}
                                      onLeaveActive={[Function]}
                                      onLeaveStart={[Function]}
                                      removeOnLeave={true}
                                      visible={false}
                                    >
                                      <DomWrapper />
                                    </CSSMotion>
                                  </LoadingIcon>
                                  <span
                                    key=".0"
                                  >
                                    Create API Token
                                  </span>
                                </button>
                              </Wave>
                            </Button>
                          </div>
                        </div>
                        <ErrorList
                          errors={Array []}
                          onDomErrorVisibleChange={[Function]}
                        >
                          <CSSMotion
                            motionAppear={true}
                            motionDeadline={500}
                            motionName="show-help"
                            onLeaveEnd={[Function]}
                            removeOnLeave={true}
                            visible={false}
                          >
                            <DomWrapper />
                          </CSSMotion>
                        </ErrorList>
                      </div>
                    </Col>
                  </FormItemInput>
                </div>
              </Row>
            </FormItem>
          </form>
        </ForwardRef(Form)>
      </SizeContextProvider>
    </ForwardRef(InternalForm)>
  </CreateTokenForm>
</Provider>
`;
